ls1intum/Artemis
# Add Kotlin pipeline to Jenkins & Sequential Test Run support

### Checklist

### Motivation and Context


It's currently not possible to generate Kotlin programming exercises with the Jenkins/Gitlab setup. Also the "sequential test runs" is not supported.

### Description

This PR does two things:
1. Adds a new Jenkinsfile for Kotlin which configures the pipeline. It allows Kotlin programming exercises to be generated in Artemis
2. Restructures the Jenkins templates directories to add support for sequential test runs
3. Adds sequential test runs support for Java and Kotlin exercises

### Steps for Testing

Since these changes affect Gitlab/Jenkins, you need to use TS2 to test.
1. Log in to Artemis
2. Navigate to Course Administration
3. Click on "Generate a new programming exercise"
4. Check that Kotlin is visible in the drop down list
5. Click on Kotlin and make sure the "sequential test run" option is clickable
5. Select the option and generate the exercise
6. You should be able to participate to the exercise, submit code, and receive feedback
7. You can also check that the feedback received is the same to the feedbacks returned in TS1 (basically reproduce the steps on TS1)
8. Generate a new Java programming exercise and make sure that the sequential test run option is also selectable
9. Click it and participate in the exercise
10. You should be able to receive feedback

### Test Coverage



JenkinsBuildPlanCreator.java 95%
JenkinsBuildPlanService.java 89%
JenkinsProgrammingLanguageFeatureService.java 100%
JenkinsXmlConfigBuilder.java  
### Screenshots


![image](https://user-images.githubusercontent.com/11298674/114534310-290d0280-9c4f-11eb-82c5-acdab4bf0280.png)

![image](https://user-images.githubusercontent.com/11298674/114534484-522d9300-9c4f-11eb-8878-c42ca172c9cc.png)
